### DISCUSSION TITLE
Prometheus Endpoint is not available

### DISCUSSION THREAD

**Discussion by chengyak** [dotansimha/envelop#1543] [Q&A] [answered]
I set the Prometheus by adding the plugin like
usePrometheus({
    // all optional, and by default, all set to false, please opt-in to the metrics you wish to get
    requestCount: true, // requries `execute` to be true as well
    requestSummary: true, // requries `execute` to be true as well
    parse: true,
    validate: true,
    contextBuilding: true,
    execute: true,
    errors: true,
    resolvers: true, // requires "execute" to be `true` as well
    resolversWhitelist: ['Mutation.*', 'Query.*'], // reports metrics als for these resolvers, leave `undefined` to report all fields
    deprecatedFields: true,
    
  })

with graph-mesh, but i cannot see the metrics in /metrics path. What is the right way to expose the metrics info to an endpoint?

**Comment by dotansimha** [comment, marked as answer]
Hi @chengyak ! 

With the pure Envelop plugin `usePrometheus`, we do not provide the service or the runtime for the Prom server. 
Since this plugin only hooks to your `prom-client` instance, you should use the same instance, and just return it over JSON response, something like:

```
server.on("GET", "/metrics", async (req, res) => {
  res.header('Content-Type', promClient.register.contentType);
  const result = await promClient.register.metrics();
  res.send(result)
}
```

In GraphQL-Mesh, we have a configuration to set this up automatically for you, so you don't need to use the raw Envelop plugin, and you should use the Mesh plugin for Prom (see https://www.the-guild.dev/graphql/mesh/docs/plugins/prometheus).

**Reply by chengyak** [reply]
Thanks for the reply @dotansimha , yeah I understand there is a plugin I can use with GraphQL-Mesh but how i can expose this endpoint with default GraphQL-Yoga? or if we need to do so, we have to use the custom server? Thanks again.

**Reply by dotansimha** [reply]
To use with Yoga v2, you'll need to expose it from your HTTP server, and implement a similar endpoint to what I shared before. 

With v3 (currently in `alpha`, we plan to rework some parts of Envelop/Yoga and introduce those plugins with a built-in HTTP setup, see https://github.com/dotansimha/graphql-yoga/issues/1732)

**Reply by chengyak** [reply]
Thanks, so if i understand it correctly, to use this in GraphQL-mesh now, i need to use custom server to expose this, in the future, there will be Envelop plug i can use to add a http endpoint to the GraphQL Yoga

**Reply by dotansimha** [reply]
Yes, but with Mesh, you can use the built-in plugin, it also exposes `/metrics` since the latest release (see https://github.com/Urigo/graphql-mesh/discussions/4645 )
